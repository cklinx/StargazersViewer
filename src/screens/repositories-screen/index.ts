export {RepositoriesScreen} from './repositories-screen';
