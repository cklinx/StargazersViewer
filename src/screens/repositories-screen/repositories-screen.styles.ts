import {StyleSheet} from 'react-native';
import {Fonts} from '@stargazers/theme';
import spacing from '@stargazers/theme/spacing';

export const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#FFF',
  },
  cellWrapper: {
    flexDirection: 'row',
    flex: 1,
  },
  scrollView: {},
  cell: {
    alignItems: 'flex-start',
    borderBottomColor: '#F4f5f7',
    borderBottomWidth: 1,
    padding: 8,
    paddingTop: 10,
    justifyContent: 'center',
    minHeight: 82,
  },
  imageWrapper: {
    position: 'relative',
    width: 50,
    height: 50,
    marginEnd: 15,
    borderWidth: 0,
    borderColor: 'rgba(0,0,0,0.1)',
    borderRadius: 50,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 5,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
  },
  badge: {
    width: 20,
    height: 20,
    backgroundColor: '#9b8af0',
    borderWidth: 0,
    borderRadius: 3,
    overflow: 'hidden',
    fontSize: 12,
    fontWeight: 'bold',
    textAlign: 'center',
    color: '#FFF',
    position: 'absolute',
    paddingTop: 2,
    textAlignVertical: 'center',
    right: 5,
    top: 10,
  },
  threadImage: {
    width: 50,
    height: 50,
    top: 0,
    borderRadius: 50,
  },
  textsWrapper: {
    fontFamily: Fonts.regular,
    overflow: 'hidden',
    flex: 1,
    shadowOffset: {
      width: 0,
      height: 0,
    },
    textShadowOffset: {width: 0, height: 0},
  },
  searchBarWrapper: {
    backgroundColor: '#FFFFFF',
    alignItems: 'center',
    justifyContent: 'center',
    height: 50,
    paddingHorizontal: spacing.s025,
    paddingVertical: spacing.s010,
  },
  searchBarText: {
    borderRadius: 25,
    borderColor: '#333',
    paddingHorizontal: 15,
    fontSize: 14,
    height: 40,
    color: 'white',
    width: '100%',
  },
  titleText: {
    fontSize: 16,
    color: '#281d6c',
    fontFamily: Fonts.semibold,
    maxWidth: '90%',
    shadowOffset: {
      width: 0,
      height: 0,
    },
    shadowOpacity: 0,
    shadowRadius: 0,
  },
  separator: {},
  productCounterBar: {
    backgroundColor: 'white',
    paddingHorizontal: 14,
    paddingVertical: 5,
  },
  productCountLabel: {
    fontSize: 15,
    color: '#FFF',
    padding: 3,
    fontFamily: Fonts.helveticaMedium,
  },
});
