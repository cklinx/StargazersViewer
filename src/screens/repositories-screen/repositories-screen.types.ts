import {ListRenderItemInfo} from 'react-native';
import {RepositoryItem} from '@stargazers/services/user-service/user-service.types';

export interface RepositoryCellProps extends ListRenderItemInfo<RepositoryItem> {
  onRepoClick: (item: RepositoryItem) => void;
}
