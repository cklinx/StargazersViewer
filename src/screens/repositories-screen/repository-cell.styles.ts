import {StyleSheet} from 'react-native';
import {Colors, Fonts} from '@stargazers/theme';
import spacing from '@stargazers/theme/spacing';

export const styles = StyleSheet.create({
  cellWrapper: {
    flexDirection: 'row',
    flex: 1,
    justifyContent: 'space-between',
  },
  cell: {
    alignItems: 'flex-start',
    borderBottomColor: '#F4f5f7',
    borderBottomWidth: 1,
    padding: 8,
    paddingVertical: 10,
    justifyContent: 'center',
    minHeight: 60,
  },
  imageWrapper: {
    justifyContent: 'center',
    alignItems: 'center',
    alignContent: 'center',
    marginEnd: 20,
  },
  avatarWrapper: {
    justifyContent: 'center',
    alignItems: 'center',
    alignContent: 'center',
    marginEnd: 20,
    borderRadius: 30,
    borderWidth: 2,
    borderColor: '#000',
    width: 60,
    height: 60,
    backgroundColor: 'white',
    overflow: 'hidden',
  },
  avatarImage: {
    width: 60,
    height: 60,
    alignSelf: 'center',
    alignItems: 'center',
  },
  placeholder: {
    resizeMode: 'cover',
    justifyContent: 'center',
    alignItems: 'center',
    width: '100%',
    height: '100%',
  },
  textsContainer: {
    fontFamily: Fonts.regular,
    overflow: 'hidden',
    flex: 1,
    shadowOffset: {
      width: 0,
      height: 0,
    },
    textShadowOffset: {width: 0, height: 0},
  },
  rightColumn: {
    alignSelf: 'center',
    justifyContent: 'center',
    alignItems: 'center',
    alignContent: 'center',
    flexDirection: 'column',
    right: 3,
    marginTop: 0,
  },
  productName: {
    fontFamily: Fonts.helveticaMedium,
    fontSize: 18,
    lineHeight: 18,
    color: '#281d6c',
    marginBottom: 2,
    paddingTop: 8,
    shadowOffset: {
      width: 0,
      height: 0,
    },
    shadowOpacity: 0,
    shadowRadius: 0,
  },
  updatedAtWrapper: {
    flexDirection: 'row',
    display: 'flex',
    justifyContent: 'flex-start',
    marginTop: 4,
    height: 18,
  },
  updatedAt: {
    fontSize: 12,
    color: '#281d6c',
    marginLeft: 0,
    lineHeight: 18,
    fontFamily: Fonts.regular,
    flex: 1,
    shadowOffset: {
      width: 0,
      height: 0,
    },
    shadowOpacity: 0,
    shadowRadius: 0,
    paddingBottom: 5,
  },
  counterRowsWrapper: {
    flexDirection: 'row',
    display: 'flex',
    justifyContent: 'flex-start',
    alignItems: 'center',
    alignContent: 'center',
    marginTop: 6,
  },
  starCounterImage: {
    width: 12,
    height: 12,
    alignSelf: 'center',
    alignItems: 'center',
    marginEnd: 5,
  },
  starCounterText: {
    marginLeft: 0,
    fontSize: 12,
    fontFamily: Fonts.regular,
  },
});
