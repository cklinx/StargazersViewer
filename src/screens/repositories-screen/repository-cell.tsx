import React from 'react';
import {Image, Text, TouchableHighlight, View} from 'react-native';
import {Colors} from '@stargazers/theme';
import {styles} from '@stargazers/screens/repositories-screen/repository-cell.styles';
import SvgArrowRight from '@stargazers/assets/Svg.ArrowRight';
import moment from 'moment';
import useColorScheme from '@stargazers/hooks/useColorScheme';
import {RepositoryCellProps} from '@stargazers/screens/repositories-screen/repositories-screen.types';

export const RepositoryCell: React.FC<RepositoryCellProps> = ({
  item,
  separators,
  onRepoClick,
}) => {
  const colorScheme = useColorScheme();

  return (
    <>
      <TouchableHighlight
        style={{
          ...styles.cell,
          borderBottomColor: Colors[colorScheme].cellBorderBottomColor,
          backgroundColor: Colors[colorScheme].background,
        }}
        underlayColor={Colors[colorScheme].cellHighlight}
        onShowUnderlay={separators.highlight}
        onHideUnderlay={separators.unhighlight}
        onPress={() => onRepoClick(item)}>
        <View style={styles.cellWrapper}>
          <View
            style={{
              ...styles.imageWrapper,
            }}>
            <Image
              source={
                colorScheme === 'dark'
                  ? require(`../../assets/repo-icon-dark.png`)
                  : require(`../../assets/repo-icon.png`)
              }
              style={{width: 24, height: 24, alignSelf: 'center', alignItems: 'center'}}
            />
          </View>
          <View style={styles.textsContainer}>
            <Text
              style={{
                ...styles.productName,
                color: Colors[colorScheme].tint,
              }}
              numberOfLines={2}
              ellipsizeMode="tail">
              {item.name}
            </Text>

            <View style={styles.updatedAtWrapper}>
              <Text
                style={{
                  ...styles.updatedAt,
                  color: Colors[colorScheme].tint,
                }}>
                {moment(item.updated_at).format('MMMM d')},{' '}
                <Text style={{fontWeight: 'bold'}}>{moment(item.updated_at).format('HH:mm')}</Text>
              </Text>
            </View>
            <View style={styles.counterRowsWrapper}>
              <Image
                source={
                  colorScheme === 'dark'
                    ? require(`../../assets/star-icon-dark.png`)
                    : require(`../../assets/star-icon.png`)
                }
                style={styles.starCounterImage}
              />
              <Text
                style={{
                  ...styles.starCounterText,
                  color: Colors[colorScheme].tint,
                }}>
                {item.stargazers_count}
              </Text>
            </View>
          </View>
          <View style={styles.rightColumn}>
            <SvgArrowRight
              size={20}
              style={{
                color: Colors[colorScheme].tint,
              }}
            />
          </View>
        </View>
      </TouchableHighlight>
    </>
  );
};
