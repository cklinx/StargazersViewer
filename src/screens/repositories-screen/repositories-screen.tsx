import React, {useEffect, useState} from 'react';

import useColorScheme from '../../hooks/useColorScheme';
import {FlatList, View, Text, Platform, RefreshControl, TextInput} from 'react-native';
import {getStargazers, getUserRepositories} from '@stargazers/services/user-service/user-service';
import {RepositoryItem} from '@stargazers/services/user-service/user-service.types';
import {styles} from './repositories-screen.styles';
import {RepositoriesScreenProps} from '@stargazers/navigators/navigator.types';
import {Colors, GlobalStyles} from '@stargazers/theme';
import {RepositoryCell} from '@stargazers/screens/repositories-screen/repository-cell';
import {useNavigation} from '@react-navigation/native';

export const RepositoriesScreen = ({route}: RepositoriesScreenProps) => {

  const colorScheme = useColorScheme();
  const owner: string = route.params.owner;
  const [refreshing, setRefreshing] = useState(false);
  const navigation = useNavigation();
  const [repositories, setRepositories] = useState<RepositoryItem[]>(route.params.repositories);
  const [filteredRepositories, setFilteredRepositories] = useState<RepositoryItem[]>(
    route.params.repositories,
  );

  useEffect(() => {
    navigation.setOptions({
      title: owner,
    });
  }, [owner]);

  const onRefresh = React.useCallback(() => {
    setRefreshing(true);
    getUserRepositories(owner).then(
      response => {
        setRepositories(response.data);
        setRefreshing(false);
      },
      () => {
        setRefreshing(false);
      },
    );
  }, []);

  const searchFilterFunction = (text: string) => {
    if (text) {
      const newData = repositories.filter(function (item) {
        const itemData = item.name ? item.name.toUpperCase() : ''.toUpperCase();
        const textData = text.toUpperCase();
        return itemData.indexOf(textData) > -1;
      });
      setFilteredRepositories(newData);
    } else {
      setFilteredRepositories(repositories);
    }
  };

  const fetchStargazers = (item: RepositoryItem) => {
    getStargazers(owner, item.name).then(
      response => {
        //@ts-ignore
        navigation.navigate('StargazersScreen', {
          owner: owner,
          repository: item,
          stargazers: response.data,
        });
      },
      () => {
        console.log('error');
      },
    );
  };

  return (
    <View
      style={{
        backgroundColor: Colors[colorScheme].background,
        flex: 1,
      }}>
      <View
        style={{
          ...styles.searchBarWrapper,
          backgroundColor: Colors[colorScheme].background,
        }}>
        <TextInput
          autoCapitalize="none"
          autoCorrect={false}
          onChangeText={searchFilterFunction}
          placeholder="Search repo.."
          placeholderTextColor={Colors[colorScheme].infoTextColor}
          style={{
            ...styles.searchBarText,
            color: Colors[colorScheme].infoTextColor,
            backgroundColor: Colors[colorScheme].secondary,
          }}
        />
      </View>
      <FlatList
        //@ts-ignore
        ItemSeparatorComponent={
          Platform.OS !== 'android' &&
          (({highlighted}) => <View style={[styles.separator, highlighted && {marginLeft: 0}]} />)
        }
        removeClippedSubviews={true}
        style={styles.scrollView}
        data={filteredRepositories}
        keyExtractor={item => item.id.toString()}
        renderItem={props => {
          return <RepositoryCell {...props} onRepoClick={fetchStargazers} />;
        }}
        refreshControl={<RefreshControl refreshing={refreshing} onRefresh={onRefresh} />}
        ListHeaderComponent={() => {
          return (
            <View
              style={{
                ...styles.productCounterBar,
                backgroundColor: Colors[colorScheme].background,
              }}>
              <Text
                style={{
                  ...styles.productCountLabel,
                  color: Colors[colorScheme].secondary,
                }}>
                {filteredRepositories.length === 1
                  ? filteredRepositories.length + ' repository'
                  : filteredRepositories.length + ' repositories'}
              </Text>
            </View>
          );
        }}
        stickyHeaderIndices={[0]}
        ListEmptyComponent={() => {
          return (
            <View style={GlobalStyles.emptyListWrapper}>
              <Text
                style={{
                  ...GlobalStyles.emptyListText,
                  color: Colors[colorScheme].tint,
                }}>
                No Repositories
              </Text>
            </View>
          );
        }}
      />
    </View>
  );
};
