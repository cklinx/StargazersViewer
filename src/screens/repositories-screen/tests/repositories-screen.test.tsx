import React from 'react';

import renderer from 'react-test-renderer';
import {RepositoriesScreen} from '../repositories-screen';
import {render} from '@testing-library/react-native';
import {FlatList} from 'react-native';
import {RepositoryCell} from '@stargazers/screens/repositories-screen/repository-cell';

jest.mock('react-native-vector-icons/Feather', () => 'Feather');
const goBack = jest.fn();
const navigate = jest.fn();
const setOptions = jest.fn();
const addListener = jest.fn();

const mockedNavigate = jest.fn();
type AppRepo = {
  id: number;
  name: string;
  updated_at: Date;
};
const mockRepositories: AppRepo[] = [
  {
    id: 1,
    name: 'Repo 1',
    updated_at: new Date(500000000000),
  },
  {
    id: 2,
    name: 'Repo 2',
    updated_at: new Date(500000000000),
  },
  {
    id: 3,
    name: 'Repo 3',
    updated_at: new Date(500000000000),
  },
];
const mockRoute = {
  params: {
    owner: 'test',
    repositories: mockRepositories,
  },
};
jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockedNavigate,
      setOptions: jest.fn(),
    }),
  };
});
describe('RepositoriesScreen', () => {
  it('should render correctly', () => {
    const {toJSON} = renderer.create(
      <RepositoriesScreen
        //@ts-ignore
        navigation={{goBack, navigate, setOptions, addListener}}
        //@ts-ignore
        route={mockRoute}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render exact cell as per source', () => {
    const componentTree = render(
      <RepositoriesScreen
        //@ts-ignore
        navigation={{goBack, navigate, setOptions, addListener}}
        //@ts-ignore
        route={mockRoute}
      />,
    );

    expect(componentTree.UNSAFE_getAllByType(FlatList).length).toBe(1);
    expect(componentTree.UNSAFE_getAllByType(RepositoryCell).length).toBe(mockRepositories.length);
  });
});
