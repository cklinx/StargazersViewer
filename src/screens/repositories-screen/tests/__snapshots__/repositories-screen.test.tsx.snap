// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoriesScreen should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "height": 50,
        "justifyContent": "center",
        "paddingHorizontal": 10,
        "paddingVertical": 4,
      }
    }
  >
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="Search repo.."
      placeholderTextColor="#FFFFFF"
      style={
        Object {
          "backgroundColor": "#414757",
          "borderColor": "#333",
          "borderRadius": 25,
          "color": "#FFFFFF",
          "fontSize": 14,
          "height": 40,
          "paddingHorizontal": 15,
          "width": "100%",
        }
      }
    />
  </View>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    ListEmptyComponent={[Function]}
    ListHeaderComponent={[Function]}
    data={
      Array [
        Object {
          "id": 1,
          "name": "Repo 1",
          "updated_at": 1985-11-05T00:53:20.000Z,
        },
        Object {
          "id": 2,
          "name": "Repo 2",
          "updated_at": 1985-11-05T00:53:20.000Z,
        },
        Object {
          "id": 3,
          "name": "Repo 3",
          "updated_at": 1985-11-05T00:53:20.000Z,
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    removeClippedSubviews={true}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={
      Array [
        0,
      ]
    }
    style={Object {}}
    viewabilityConfigCallbackPairs={Array []}
  >
    <RCTRefreshControl />
    <View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "paddingHorizontal": 14,
              "paddingVertical": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#414757",
                "fontFamily": "HelveticaNeue-Medium",
                "fontSize": 15,
                "padding": 3,
              }
            }
          >
            3 repositories
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "flex-start",
              "backgroundColor": "#fff",
              "borderBottomColor": "#f4f4f4",
              "borderBottomWidth": 1,
              "justifyContent": "center",
              "minHeight": 60,
              "padding": 8,
              "paddingVertical": 10,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "justifyContent": "center",
                  "marginEnd": 20,
                }
              }
            >
              <Image
                source="test-file-stub"
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": 24,
                    "width": 24,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "fontFamily": "Montserrat-Regular",
                  "overflow": "hidden",
                  "shadowOffset": Object {
                    "height": 0,
                    "width": 0,
                  },
                  "textShadowOffset": Object {
                    "height": 0,
                    "width": 0,
                  },
                }
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={2}
                style={
                  Object {
                    "color": "#383B42",
                    "fontFamily": "HelveticaNeue-Medium",
                    "fontSize": 18,
                    "lineHeight": 18,
                    "marginBottom": 2,
                    "paddingTop": 8,
                    "shadowOffset": Object {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                  }
                }
              >
                Repo 1
              </Text>
              <View
                style={
                  Object {
                    "display": "flex",
                    "flexDirection": "row",
                    "height": 18,
                    "justifyContent": "flex-start",
                    "marginTop": 4,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#383B42",
                      "flex": 1,
                      "fontFamily": "Montserrat-Regular",
                      "fontSize": 12,
                      "lineHeight": 18,
                      "marginLeft": 0,
                      "paddingBottom": 5,
                      "shadowOffset": Object {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0,
                      "shadowRadius": 0,
                    }
                  }
                >
                  November 2
                  ,
                   
                  <Text
                    style={
                      Object {
                        "fontWeight": "bold",
                      }
                    }
                  >
                    01:53
                  </Text>
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignContent": "center",
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "marginTop": 6,
                  }
                }
              >
                <Image
                  source="test-file-stub"
                  style={
                    Object {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "height": 12,
                      "marginEnd": 5,
                      "width": 12,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "color": "#383B42",
                      "fontFamily": "Montserrat-Regular",
                      "fontSize": 12,
                      "marginLeft": 0,
                    }
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "alignSelf": "center",
                  "flexDirection": "column",
                  "justifyContent": "center",
                  "marginTop": 0,
                  "right": 3,
                }
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="20"
                bbWidth="20"
                focusable={false}
                height={20}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {
                      "color": "#383B42",
                    },
                    Object {
                      "flex": 0,
                      "height": 20,
                      "width": 20,
                    },
                  ]
                }
                tintColor="#383B42"
                vbHeight={24}
                vbWidth={24}
                width={20}
              >
                <RNSVGGroup>
                  <RNSVGPath
                    d="M15.7 11.3l-6-6c-.4-.4-1-.4-1.4 0s-.4 1 0 1.4l5.3 5.3-5.3 5.3c-.4.4-.4 1 0 1.4.2.2.4.3.7.3s.5-.1.7-.3l6-6c.4-.4.4-1 0-1.4z"
                    fill={
                      Object {
                        "payload": 4281875266,
                        "type": 0,
                      }
                    }
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {},
              false,
            ]
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "flex-start",
              "backgroundColor": "#fff",
              "borderBottomColor": "#f4f4f4",
              "borderBottomWidth": 1,
              "justifyContent": "center",
              "minHeight": 60,
              "padding": 8,
              "paddingVertical": 10,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "justifyContent": "center",
                  "marginEnd": 20,
                }
              }
            >
              <Image
                source="test-file-stub"
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": 24,
                    "width": 24,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "fontFamily": "Montserrat-Regular",
                  "overflow": "hidden",
                  "shadowOffset": Object {
                    "height": 0,
                    "width": 0,
                  },
                  "textShadowOffset": Object {
                    "height": 0,
                    "width": 0,
                  },
                }
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={2}
                style={
                  Object {
                    "color": "#383B42",
                    "fontFamily": "HelveticaNeue-Medium",
                    "fontSize": 18,
                    "lineHeight": 18,
                    "marginBottom": 2,
                    "paddingTop": 8,
                    "shadowOffset": Object {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                  }
                }
              >
                Repo 2
              </Text>
              <View
                style={
                  Object {
                    "display": "flex",
                    "flexDirection": "row",
                    "height": 18,
                    "justifyContent": "flex-start",
                    "marginTop": 4,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#383B42",
                      "flex": 1,
                      "fontFamily": "Montserrat-Regular",
                      "fontSize": 12,
                      "lineHeight": 18,
                      "marginLeft": 0,
                      "paddingBottom": 5,
                      "shadowOffset": Object {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0,
                      "shadowRadius": 0,
                    }
                  }
                >
                  November 2
                  ,
                   
                  <Text
                    style={
                      Object {
                        "fontWeight": "bold",
                      }
                    }
                  >
                    01:53
                  </Text>
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignContent": "center",
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "marginTop": 6,
                  }
                }
              >
                <Image
                  source="test-file-stub"
                  style={
                    Object {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "height": 12,
                      "marginEnd": 5,
                      "width": 12,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "color": "#383B42",
                      "fontFamily": "Montserrat-Regular",
                      "fontSize": 12,
                      "marginLeft": 0,
                    }
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "alignSelf": "center",
                  "flexDirection": "column",
                  "justifyContent": "center",
                  "marginTop": 0,
                  "right": 3,
                }
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="20"
                bbWidth="20"
                focusable={false}
                height={20}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {
                      "color": "#383B42",
                    },
                    Object {
                      "flex": 0,
                      "height": 20,
                      "width": 20,
                    },
                  ]
                }
                tintColor="#383B42"
                vbHeight={24}
                vbWidth={24}
                width={20}
              >
                <RNSVGGroup>
                  <RNSVGPath
                    d="M15.7 11.3l-6-6c-.4-.4-1-.4-1.4 0s-.4 1 0 1.4l5.3 5.3-5.3 5.3c-.4.4-.4 1 0 1.4.2.2.4.3.7.3s.5-.1.7-.3l6-6c.4-.4.4-1 0-1.4z"
                    fill={
                      Object {
                        "payload": 4281875266,
                        "type": 0,
                      }
                    }
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {},
              false,
            ]
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "flex-start",
              "backgroundColor": "#fff",
              "borderBottomColor": "#f4f4f4",
              "borderBottomWidth": 1,
              "justifyContent": "center",
              "minHeight": 60,
              "padding": 8,
              "paddingVertical": 10,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "justifyContent": "center",
                  "marginEnd": 20,
                }
              }
            >
              <Image
                source="test-file-stub"
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": 24,
                    "width": 24,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "fontFamily": "Montserrat-Regular",
                  "overflow": "hidden",
                  "shadowOffset": Object {
                    "height": 0,
                    "width": 0,
                  },
                  "textShadowOffset": Object {
                    "height": 0,
                    "width": 0,
                  },
                }
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={2}
                style={
                  Object {
                    "color": "#383B42",
                    "fontFamily": "HelveticaNeue-Medium",
                    "fontSize": 18,
                    "lineHeight": 18,
                    "marginBottom": 2,
                    "paddingTop": 8,
                    "shadowOffset": Object {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                  }
                }
              >
                Repo 3
              </Text>
              <View
                style={
                  Object {
                    "display": "flex",
                    "flexDirection": "row",
                    "height": 18,
                    "justifyContent": "flex-start",
                    "marginTop": 4,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#383B42",
                      "flex": 1,
                      "fontFamily": "Montserrat-Regular",
                      "fontSize": 12,
                      "lineHeight": 18,
                      "marginLeft": 0,
                      "paddingBottom": 5,
                      "shadowOffset": Object {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0,
                      "shadowRadius": 0,
                    }
                  }
                >
                  November 2
                  ,
                   
                  <Text
                    style={
                      Object {
                        "fontWeight": "bold",
                      }
                    }
                  >
                    01:53
                  </Text>
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignContent": "center",
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "marginTop": 6,
                  }
                }
              >
                <Image
                  source="test-file-stub"
                  style={
                    Object {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "height": 12,
                      "marginEnd": 5,
                      "width": 12,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "color": "#383B42",
                      "fontFamily": "Montserrat-Regular",
                      "fontSize": 12,
                      "marginLeft": 0,
                    }
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "alignSelf": "center",
                  "flexDirection": "column",
                  "justifyContent": "center",
                  "marginTop": 0,
                  "right": 3,
                }
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="20"
                bbWidth="20"
                focusable={false}
                height={20}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {
                      "color": "#383B42",
                    },
                    Object {
                      "flex": 0,
                      "height": 20,
                      "width": 20,
                    },
                  ]
                }
                tintColor="#383B42"
                vbHeight={24}
                vbWidth={24}
                width={20}
              >
                <RNSVGGroup>
                  <RNSVGPath
                    d="M15.7 11.3l-6-6c-.4-.4-1-.4-1.4 0s-.4 1 0 1.4l5.3 5.3-5.3 5.3c-.4.4-.4 1 0 1.4.2.2.4.3.7.3s.5-.1.7-.3l6-6c.4-.4.4-1 0-1.4z"
                    fill={
                      Object {
                        "payload": 4281875266,
                        "type": 0,
                      }
                    }
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
